cmd/dep/status.go

```diff
@@ -722,6 +732,234 @@ outer:
 	return hasMissingPkgs, 0, errInputDigestMismatch
 }
 
+func runOld(ctx *dep.Ctx, out outputter, p *dep.Project, sm gps.SourceManager) (hasMissingPkgs bool, errCount int, err error) {
+	ptree, err := p.ParseRootPackageTree()
+	if err != nil {
+		return false, 0, err
+	}
+
+	// Solve the dependencies so we can check if the current dependencies are old
+	params := gps.SolveParameters{
+		ProjectAnalyzer: dep.Analyzer{},
+		RootDir:         p.AbsRoot,
+		RootPackageTree: ptree,
+		Manifest:        p.Manifest,
+		ChangeAll:       true,
+	}
+	s, err := gps.Prepare(params, sm)
+	if err != nil {
+		return false, 0, errors.Wrapf(err, "could not set up solver for input hashing")
+	}
+
+	if err := out.BasicHeader(); err != nil {
+		return false, 0, err
+	}
+
+	// Errors while collecting constraints should not fail the whole status run.
+	// It should count the error and tell the user about incomplete results.
+	cm, ccerrs := collectConstraints(ctx, p, sm)
+	if len(ccerrs) > 0 {
+		errCount += len(ccerrs)
+	}
+
+	bsMap := make(map[string]*BasicStatus)
+
+	slp := p.Lock.Projects()
+
+	if !bytes.Equal(s.HashInputs(), p.Lock.SolveMeta.InputsDigest) {
```

Comment:
We need not check this as well. We are just checking for new versions of what we have in the lock, under the set constraints.